### DISCUSSION TITLE
How to translate Qalculate! Qt UI

### DISCUSSION THREAD

**Discussion by bubuleur** [Qalculate/qalculate-qt#166] [Q&A]
Hello
I would like to come to you and I would like to know how I can translate Qalculate! Qt UI I don't know any software to translate a .qm file could you help me, I have already translated libqalculate and qalculate-gtkqalculate-gtk
in French with poedit
Thanks

**Comment by hanna-kn** [comment]
Qt Linguist (installed as part of Qt library on Windows, qt6-tools on Arch Linux, and linguist-qt6 on Ubuntu).

**Comment by bubuleur** [comment]
Thank you very much hanna-kn for your answer, I looked at this version it would be good for me ?
https://github.com/lelegard/qtlinguist-installers/releases/tag/v5.12.0 
it's for a windows version that I need

**Comment by bubuleur** [comment]
I'm trying Qt Linguist but it's much more difficult to use than poedit. Do you know how to translate with this software?

**Reply by hanna-kn** [reply]
Open [qalculate-qt_fr.ts](https://raw.githubusercontent.com/Qalculate/qalculate-qt/refs/heads/main/translations/qalculate-qt_fr.ts) in Qt Linguist. Select an object in the list on the left and a text string in middle top list. Enter the translated text string below and press `Ctrl + Enter` to mark it as finished and move to the next. You can move to the next _unfinished_ translation using `Ctrl + J`.

To reduce the necessary work, I've updated the Qt translation using the updated French qalculate-gtk translation (this process is partially automated).

**Comment by bubuleur** [comment]
I found thank you, I managed to update most of the words and sentences but I have a part which is grayed out on Qt Linguist
attached photo
![2024-11-09_17-09-11](https://github.com/user-attachments/assets/f53c97e6-9b50-4be6-b00d-baf81ed11a57)
I can't change it, is this normal ?
I think it's gone which remains to be translated because when I generate the qt_fr.qm file afterwards when I look in the software the translation has not changed

**Reply by hanna-kn** [reply]
Those are translations imported from qalculate-gtk with no corresponding text strings in the Qt UI, and grayed out items in other places are translations of text strings which has disappeared from the code.

The translations are compiled in the qalculate-qt(.exe) binary. qt_fr.qm file only contains translations provided by the Qt library.

**Comment by bubuleur** [comment]
OK if I understand correctly it is by creating the application that the translation will be done 
if so, is it possible to have the file or something else to see if the translation was done correctly?

**Reply by hanna-kn** [reply]
A recompiled binary is now available at https://github.com/Qalculate/qalculate-qt/releases/download/v5.3.0/qalculate-qt-exe-fr-test.zip.

**Comment by bubuleur** [comment]
thank you very much for your file I noticed that it is missing in untranslated words 
![2024-11-10_16-05-40](https://github.com/user-attachments/assets/e3279db7-3a41-42d1-a18f-a5565ded22f5)
Afterwards I don't know if there are others I looked it seems to me that no but I'm not sure
thanking you

**Reply by hanna-kn** [reply]
The untranslated text was caused by a mistake in the code. After this has been fixed (https://github.com/Qalculate/qalculate-qt/commit/d7e8b9532abbc2e1f60a0abd477f5ae5a19213a5), the only untranslated items, in the French translation, are "Use dot as multiplication sign" and "Use Unicode division slash in output".

Here is the updated preferences dialog:
![Skärmbild_20241110_163430](https://github.com/user-attachments/assets/8b130196-469f-462c-87c4-43af13a254b3)

**Comment by bubuleur** [comment]
ok thank you for your answer
it is possible to have an exe file to see also with the sentences "Use dot as multiplication sign" and "Use Unicode division slash in output" translated into French?
thank you for all

**Reply by hanna-kn** [reply]
I've now updated https://github.com/Qalculate/qalculate-qt/releases/download/v5.3.0/qalculate-qt-exe-fr-test.zip.

**Comment by bubuleur** [comment]
it's perfect thank you very much for everything
How soon will you release a new version?

**Reply by hanna-kn** [reply]
2024-11-25 or 2024-12-02
